components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2130,20 +2130,46 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    async getStripePortalUrlForTeam(ctx: TraceContext, teamId: string): Promise<string> {
-        this.checkAndBlockUser("getStripePortalUrlForTeam");
-        const team = await this.guardTeamOperation(teamId, "update");
-        await this.ensureStripeApiIsAllowed({ team });
-        try {
-            const url = await this.stripeService.getPortalUrlForTeam(team!);
-            return url;
-        } catch (error) {
-            log.error(`Failed to get Stripe portal URL for team '${teamId}'`, error);
-            throw new ResponseError(
-                ErrorCodes.INTERNAL_SERVER_ERROR,
-                `Failed to get Stripe portal URL for team '${teamId}'`,
-            );
+    async getStripePortalUrl(ctx: TraceContext, attributionId: string): Promise<string> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
+
+        const user = this.checkAndBlockUser("getStripePortalUrl");
+
+        let url: string;
+        if (attrId.kind === "user") {
+            await this.ensureStripeApiIsAllowed({ user });
+            try {
+                url = await this.stripeService.getPortalUrlForUser(user);
+            } catch (error) {
+                log.error(`Failed to get Stripe portal URL for user '${user.id}'`, error);
+                throw new ResponseError(
+                    ErrorCodes.INTERNAL_SERVER_ERROR,
+                    `Failed to get Stripe portal URL for user '${user.id}'`,
+                );
+            }
+        } else {
+            const team = await this.guardTeamOperation(attrId.teamId, "update");
+            await this.ensureStripeApiIsAllowed({ team });
+            try {
+                url = await this.stripeService.getPortalUrlForTeam(team);
+            } catch (error) {
+                log.error(`Failed to get Stripe portal URL for team '${team.id}'`, error);
+                throw new ResponseError(
+                    ErrorCodes.INTERNAL_SERVER_ERROR,
+                    `Failed to get Stripe portal URL for team '${team.id}'`,
+                );
+            }
         }
+        return url;
+    }
+
+    async getStripePortalUrlForTeam(ctx: TraceContext, teamId: string): Promise<string> {
+        const attributionId: AttributionId = { kind: "team", teamId: teamId };
+        return this.getStripePortalUrl(ctx, AttributionId.render(attributionId));
```

Comment:
Side note: You could easily remove this now-somewhat-useless method by updating the single call-site in the dashboard.

Reply:
I'll move the dashboard over to the new more general methods in a separate PR later.